### DISCUSSION TITLE
Ability to limit the max number of lines per match when -U is used

### DISCUSSION THREAD

**Discussion by amosbird** [BurntSushi/ripgrep#2093] [Q&A] [answered]
This is a continuation of the discussion in https://github.com/BurntSushi/ripgrep/issues/176#issuecomment-983680281

Amos: Do we have an option to limit the max number of lines each match can have?

Andrew: No. And I don't see any obvious way to implement that either. You can usually build such limits into your regex instead.

Amos: Can we build a regex to match the following? `foo.*[at most three new lines]bar.*[at most three new lines]baz.*` (**with all together at most three new lines**)

Andrew: Sure? Why not? `foo(.*\n?){0,3}bar(.*\n?){0,3}`, or something like that anyway.

@BurntSushi  But how can `foo(.\n?){0,3}bar(.\n?){0,3}` limit the total number of lines smaller than 3? It only limits each.

**Comment by BurntSushi** [comment, marked as answer]
Ah I see. In that case, I suspect the only thing you can do is some sort of look-around magic with the PCRE2 engine. But I'm not an expert in such things, so you'd have better luck asking elsewhere. At that point, it's not really a ripgrep specific question any more.

I don't really think there is much else that can be done here.

**Reply by amosbird** [reply]
Ok. I'll investigate. Thanks!
